### PR TITLE
Fix auto-reconnect

### LINKED ISSUE #39: Auto-reconnect doesn't work properly
The ansq connection re-connects only once and if it fails it would stop `nsq.messages()` generator immediately - which most like cause to stop consumer service. The connection with enabled auto-reconnect feature should try to reconnect each N seconds (or progressively) and wait until the connection is established.

### PR DESCRIPTION
**Pull request by atugushev**
Fixes #39 

TODO:

- [x] wait for #38 and write tests

### CODE REVIEW

File: ansq/tcp/connection.py [hunk 1]
```diff
@@ -476,6 +500,7 @@ async def open_connection(
     on_exception: Callable = None,
     loop: AbstractEventLoop = None,
     auto_reconnect: bool = True,
+    auto_reconnect_interval: int = 15000,
```
Comment (atugushev) [inline ansq/tcp/connection.py:503]: I would use `auto_reconnect_interval= 15s`, but made it to be consistent with `heartbeat_interval`.

Comment (lig) [reply ansq/tcp/connection.py:503]: IMO, in many usecases 15 seconds is too long to wait for a reconnect. On the other hand something like 1s is another extreme. In ideal world I'd like to see a progressive interval. Something like a geometric progression. Say, 2s, 4s, 8s, 16s and so on for consequent retries. Maybe up to 32s at max.

Also, the auto reconnect behavior is extensive for a lookupd supported connections. Thus, it applies for a small single instance setups only where one doesn't usually wish to spend a lot of time on tunning.

Comment (atugushev) [reply ansq/tcp/connection.py:503]: Two different approach is described [here](https://nsq.io/clients/building_client_libraries.html#reconnection) by the way:
- for a single connection there should be progressive intervals
- for a consumer with dynamic connections (lookupd) there should be a constant interval

what would you think?

**Comment by lig** [comment]
Sounds reasonable. There could be indeed cases when reconnecting is
justified regardless of data received from lookups.

ср, 23 мар. 2022 г., 15:57 Albert Tugushev ***@***.***>:

File: ansq/tcp/connection.py (hunk 1 above)
Comment (lig) [reply ansq/tcp/connection.py:503]: Sounds reasonable. There could be indeed cases when reconnecting is justified regardless of data received from lookups.

Comment (atugushev) [reply ansq/tcp/connection.py]: I've added progression.

**Pull request merged**